### PR TITLE
Fix a bug with skipping two lines after TITLE

### PR DESCRIPTION
**Pull request by eivindjahren**
**Issue**

The behavior of skipping two non-blank lines after TITLE was inherited from the code in resdata. After investigating how this is actually implemented in reservoir simulators, it turns out that one line, regardless of whether it is blank should be skipped. It seems that a blank title might actually cause a crash regardless of what comes after.

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/config/_get_num_cpu.py [hunk 1]
```diff
@@ -110,14 +110,10 @@ def _get_num_cpu(
...
-                # Skip two non-blank lines following a TITLE
-                for _ in range(2):
-                    line: list[str] = []
-                    while line == []:
-                        nline = parser.next_line(None)
-                        if nline is None:
-                            break
-                        line = list(nline)
+                # Skip one line following a TITLE
+                for _ in range(1):
```
Comment (berland) [inline src/ert/config/_get_num_cpu.py:114]: Can't this line with `for` just be dropped?

File: tests/ert/unit_tests/config/test_num_cpu.py [hunk 2]
```diff
@@ -43,18 +42,23 @@ def test_num_cpu_from_data_file_used_if_config_num_cpu_not_set(parallelsuffix):
     with open(file=data_file, mode="w", encoding="utf-8") as data_file_hander:
         data_file_hander.write(
             f"""
+RUNSPEC
+--comment
+TITLE
+-- A Test Case --
```
Comment (berland) [inline tests/ert/unit_tests/config/test_num_cpu.py:48]: would suggest not to use commenting symbols in this example line

Comment (berland) [reply tests/ert/unit_tests/config/test_num_cpu.py:48]: we should possibly parametrize the title to "", "\n", "-- foo", "foo" etc.

File: src/ert/config/_get_num_cpu.py (hunk 1 above)
Comment (eivindjahren) [reply src/ert/config/_get_num_cpu.py:114]: fixed

File: tests/ert/unit_tests/config/test_num_cpu.py [hunk 3]
```diff
@@ -22,39 +20,50 @@ def test_num_cpu_from_config_preferred():
...
+        ).preferred_num_cpu
+        == config_num_cpu
+    )
 
 
 @pytest.mark.parametrize(
     "parallelsuffix", [("/"), (" /"), (" DISTRIBUTED/"), (" DISTRIBUTED /")]
 )
+@pytest.mark.parametrize(
+    "casetitle", ["CASE", "-- A CASE --", "PARALLEL Tutorial Case"]
```
Comment (berland) [inline tests/ert/unit_tests/config/test_num_cpu.py:39]: Including PARALLEL here is a nice touch! 👍🏻

Comment (berland) [reply tests/ert/unit_tests/config/test_num_cpu.py:39]: Out code probably allows an empty line also, should we include it in the tests but comment that ECLIPSE itself actually crashes on it?

File: tests/ert/unit_tests/config/test_num_cpu.py (hunk 2 above)
Comment (eivindjahren) [reply tests/ert/unit_tests/config/test_num_cpu.py]: Added a parametrized list of titles as per our conversation.

**Review by berland — APPROVED** [review]
yay!

**Pull request merged**